spec/classes/hiera_spec.rb

```diff
@@ -59,13 +59,16 @@
           merge_behavior: 'deeper',
           eyaml: true,
           datadir: '/etc/puppetlabs/code/environments/%{::environment}/hieradata',
-          backends: %w[yaml eyaml json yamll],
+          backends: %w[yaml eyaml json yamll redis],
           'backend_options' => {
             'json' => {
               'datadir' => '/etc/puppet/json_data/data'
             },
             'yamll' => {
               'datadir' => '/etc/puppet/yamll_data/data'
+            },
+            'redis' => {
+              'deserialize' => '!ruby/sym json'
             }
```

Comment:
is adding this backend the reason for this pr?
it's not clear from the title and I'm so confused about half of these changes

Reply:
The reason behind the changes is because the regex to parse symbols is not working properly. This backend is a test case (that also is mentioned in the readme). When I was actually trying to use the mechanism to convert the value into a symbol, the output of this '!ruby/sym json' was becoming this ':undef'. The changes were only on the template file, everything else was because the CI was requesting formating to pass, I can revert them, but you will need the CI or accept without a green ack.